source/extensions/quic_listeners/quiche/platform/flags_impl.cc

```diff
@@ -0,0 +1,99 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/flags_impl.h"
+
+#include <set>
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/numbers.h"
+
+namespace quiche {
+
+// static
+FlagRegistry& FlagRegistry::GetInstance() {
+  static auto* instance = new FlagRegistry();
+  return *instance;
+}
+
+void FlagRegistry::RegisterFlag(const char* name, Flag* flag) {
+  absl::MutexLock lock(&mutex_);
+  flags_.emplace(std::string(name), flag);
+}
+
+void FlagRegistry::ResetFlags() const {
+  absl::MutexLock lock(&mutex_);
+  for (auto& kv : flags_) {
+    kv.second->ResetValue();
+  }
+}
+
+Flag* FlagRegistry::FindFlag(const std::string& name) const {
+  absl::MutexLock lock(&mutex_);
+  auto it = flags_.find(name);
+  return (it != flags_.end()) ? it->second : nullptr;
+}
+
+template <> bool TypedFlag<bool>::SetValueFromString(const std::string& value_str) {
+  static const auto* kTrueValues = new std::set<std::string>({"1", "t", "true", "y", "yes"});
+  static const auto* kFalseValues = new std::set<std::string>({"0", "f", "false", "n", "no"});
+  auto lower = absl::AsciiStrToLower(value_str);
+  if (kTrueValues->find(lower) != kTrueValues->end()) {
+    absl::MutexLock lock(&mutex_);
+    value_ = true;
+    return true;
+  }
+  if (kFalseValues->find(lower) != kFalseValues->end()) {
+    absl::MutexLock lock(&mutex_);
+    value_ = false;
+    return true;
+  }
+  return false;
+}
+
+template <> bool TypedFlag<int32_t>::SetValueFromString(const std::string& value_str) {
+  int32_t value;
+  if (absl::SimpleAtoi(value_str, &value)) {
+    absl::MutexLock lock(&mutex_);
```

Comment:
nit: How about "SetValue(value)" and "return true"? Same for other types.

Reply:
Right, of course :-) Done.